### ISSUE TITLE
[FEATURE] Enable Segment Replacation by default for k-NN indices

### ISSUE THREAD

**Issue by vamshin** [opensearch-project/k-NN#1052]
**Is your feature request related to a problem?**
k-NN graph construction is compute heavy and consumes resources. By enabling segment replication, we could cut down the cut by half by not having to construct the graphs in replica but just copy the already constructed graph

**Comment by jmazanec15** [comment]
Yes, agree this could save a lot of compute. Can we recommend users set the setting in documentation instead of making it default for k-NN?

**Comment by vamshin** [comment]
If it can deterministically save compute, why not enable it by default. Are there any cases it becomes overhead?

**Comment by jmazanec15** [comment]
Compute savings will come at the cost of more expensive networking operations - more data will be transferred over the wire. So this could be a surprise to users upgrading and potentially cause degradations in some cases. A user who is using a roll-over policy, for instance, might notice this upon upgrade.

**Comment by vamshin** [comment]
Makes sense. But I feel saving compute is more beneficial especially cutting down the compute by 50% vs  increased network traffic. Also for vector related workloads, customers usually index at once and then search, so I am seeing more benefits with compute than network. If we think this is kind of breaking change interms of experience, then we can debate about moving this default behavior to Major version release 3.x

**Comment by kotwanikunal** [comment]
@vamshin Do we still want to go this route? We can have index setting overrides as an easy solution.

**Comment by vamshin** [comment]
Sounds good. We should be good to close.
